### ISSUE TITLE
Custom loss function leads to -inf loss

### ISSUE THREAD

**Issue by code-ball** [fchollet/keras#2508]
hi,

I have my model compile with multiple loss functions including custom loss functions declared as follows:

> model.compile(optimizer='rmsprop', loss=['categorical_crossentropy','custom_loss'])

There are two problems I am facing:

1.) If I define this function locally inside my code file, I get 'invalid loss function' error. But if I change objective.py in the lib and add the custom function to it, the model is able to use it. Is this normal or is it supposed to be recognized even when I define it locally? Please let me know if I am missing something.

2.) When the model uses the function, it provides -inf values. Is there a way to debug why the loss is returned as -inf? I am sure that this custom loss function is causing the whole loss to be -inf. If either I remove the custom loss or change the definition of custom loss to something simple, it does not give -inf. 

Thanks

**Comment by code-ball** [comment]
@braingineer : Any idea on this one?

Thanks

**Comment by braingineer** [comment]
re: infs
Usually infs are the result of super big numbers, so you must have an explosion somewhere.  Theano has NanGuardMode that lets you debug these types of problems. It's super helpful.   If you're doing anything that might have 0s and those 0s could get used incorrectly, It's good to clip your numbers to a small number (1e-8 or something).  

re: custom loss. 
if you are passing a function into loss rather than the string, it should work.  When you have it locally, are you passing the string?  The string is used to look up functions in a similar way that `getattr` looks up functions.  But the 'object' it looks at it is the objectives file.   So, if the string doesn't match a function name in that file, it throws errors.   The function that does this is is at the bottom of the file (not only in objectives, but in most of the files).  If you follow that function to `keras.utils.generic_utils.get_from_module` you'll see that it checks "if string, elif dict, else return". so if you pass a function, it should transparently pass it through. 

aka, 

```python
model.compile(optimizer='rmsprop', loss=['categorical_crossentropy', my_loss_func])
```

**Comment by carlthome** [comment]
Are you using lambprop by any chance?
